File: VimSharp/chrome-extension-sharp/src/t1-sqlts/Expressions/SqlConstraintDefaultValue.ts
import { SqlType } from './SqlType';
import { TextSpan } from '../StringParser';
import { ISqlConstraint, ISqlExpression, SqlVisitor } from './ISqlExpression';

export class SqlConstraintDefaultValue implements ISqlConstraint {
    SqlType: SqlType = SqlType.ConstraintDefaultValue;
    Span: TextSpan = new TextSpan();
    ConstraintName: string = '';
    DefaultValue: string = '';

    Accept(visitor: SqlVisitor): void {
        visitor.Visit_ConstraintDefaultValue(this);
    }

    ToSql(): string {
        return `DEFAULT ${this.DefaultValue}`;
    }
} 

File: VimSharp/chrome-extension-sharp/src/t1-sqlts/Expressions/SqlConstraintDefaultValue.cs
namespace T1.SqlSharp.Expressions;

public class SqlConstraintDefaultValue : ISqlConstraint
{
    public SqlType SqlType { get; } = SqlType.ConstraintDefaultValue;
    public TextSpan Span { get; set; } = new();
    public void Accept(SqlVisitor visitor)
    {
        visitor.Visit_ConstraintDefaultValue(this);
    }

    public string ConstraintName { get; set; } = string.Empty;
    public string DefaultValue { get; set; } = string.Empty;

    public string ToSql()
    {
        return $"DEFAULT {DefaultValue}";
    }
}